shell/platform/darwin/macos/BUILD.gn

```diff
@@ -315,7 +315,15 @@ action("flutter_framework") {
     "--location",
     rebase_path(root_out_dir),
   ]
-  deps = [ ":_generate_symlinks_and_verify_framework_module" ]
+  deps = [
+    ":_generate_symlinks_and_verify_framework_module",
+    ":copy_dylib",
+    ":copy_framework_headers",
+    ":copy_framework_icu",
+    ":copy_framework_info_plist",
+    ":copy_framework_module_map",
+    ":copy_license",
```

Comment:
Since `_generate_symlinks_and_verify_framework_module` deps on `_generate_symlinks` which deps on this same list, it seems like maybe the deps aren't the source of the problem.  Maybe `sources = [ $root_build_dir/$_flutter_framework_name.stamp ]` would make this re-run if the framework is rebuilt?

@zanderso what's the gn idiomatic way to do this?

Reply:
I believe the right way to do this is to leave this deps list unchanged, but to make `_generate_symlinks` a `public_deps` of `_generate_symlinks_and_verify_framework_module`. (I think that's all that should be needed, but making the `deps` of `_generate_symlinks` be `public_deps` in addition would be the next thing to try.)

Reply:
Unfortunately setting `deps` to `public_deps` for both `_generate_symlinks` and `_generate_symlinks_and_verify_framework_module` didn't work, perhaps because `_generate_symlinks_and_verify_framework_module` is a `shared_library`?

Looking at how we do this for iOS, we have the `flutter_framework` depend on a `group` `universal_flutter_framework`, which then has a dependency on the `shared_library` (`copy_and_verify_framework_module`) and the other deps
https://github.com/flutter/engine/blob/76f33eb0ecedee4182215923e5e3942a5dfbfcc2/shell/platform/darwin/ios/BUILD.gn#L448-L478

I could add a `group` to the macOS version to mimic it, also perhaps instead of the deps from `_generate_symlinks`, just use `_generate_symlinks` as the dep. What do you think?

Reply:
@jmagman @zanderso friendly ping